pkg/client/cli/telepresence_test.go

```diff
@@ -803,6 +802,37 @@ func (cs *connectedSuite) TestM_AutoInjectedAgent() {
 	})
 }
 
+func (cs *connectedSuite) TestM_AutoInjectedUninstall() {
+	ctx := dlog.NewTestContext(cs.T(), false)
+
+	cs.NoError(cs.tpSuite.applyApp(ctx, "echo-auto-inject", "echo-auto-inject", 80))
+	defer func() {
+		cs.NoError(cs.tpSuite.kubectl(ctx, "delete", "svc,deploy", "echo-auto-inject", "--context", "default"))
+	}()
+
+	cs.Run("can be uninstalled", func() {
+		require := cs.Require()
+
+		cs.Eventually(func() bool {
+			stdout, stderr := telepresence(cs.T(), "list", "--namespace", cs.ns(), "--agents")
+			cs.Empty(stderr)
+			return strings.Contains(stdout, "echo-auto-inject: ready to intercept (traffic-agent already installed)")
+		},
+			10*time.Second, // waitFor
+			2*time.Second,  // polling interval
+		)
+
+		stdout, stderr := telepresence(cs.T(), "uninstall", "-e")
+		require.Empty(stderr)
+		require.Contains(stdout, "Root Daemon quitting... done")
+
+		stdout, stderr = telepresence(cs.T(), "list", "--namespace", cs.ns())
+		require.Empty(stderr)
+		require.Contains(stdout, "echo-auto-inject: ready to intercept (traffic-agent not yet installed)")
```

Comment:
My hunch is that maybe it's taking too long for all the agents to be uninstalled and when you call the list, the new traffic-manager is seeing the old `echo-auto-inject` agent before it's terminating.. 

One thing you could do to validate if that's the issue is between the uninstall and the list, do something like what we do here: https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/client/cli/telepresence_test.go#L1161-L1181 to ensure there's only pod (and maybe even ensure there's only one pod with one container, so we know the agent has been removed)

Reply:
Additionally, some of `TestZ_Uninstall` is affected by doing an `uninstall --everything` earlier in the tests, so perhaps it's worth moving this to the end of `TestZ_Uninstall` so we're still testing the individual uninstalls.. Or you could restore the necessary agents for those tests as part of the cleanup of this test, up to you!

Reply:
Thanks for looking it over